### DISCUSSION TITLE
Toolbar for Searchbox / Included Verticals?

### DISCUSSION THREAD

**Discussion by edarroudi** [microsoft-search/pnp-modern-search#2375] [Ideas]
Hi,

my customer had a request for adding a toolbar menu to the searchbox. As it wasn't possible with the current searchbox webpart I created another webpart that attaches itself to the searchbox. That Webpart can be configured like the vertical Webpart and additonaly provides some other features like opening a dialog/IframeDialog or execute a custom script from another extension library.
As the attaching to the webpart is a "hack" and I don't like those in the long term - I was thinking about adding the code to the searchbox webpart.

![image](https://user-images.githubusercontent.com/935493/191451866-e338823e-6a63-4071-a042-89f7f430c645.png)
![image](https://user-images.githubusercontent.com/935493/191452465-0496c41e-acba-4cb9-8b51-2bd651702116.png)
![image](https://user-images.githubusercontent.com/935493/191453274-9edbcffe-b591-4758-9209-fb9588f43476.png)
![image](https://user-images.githubusercontent.com/935493/191453666-272f7924-f067-431a-b75c-647de1442f12.png)

It doesn't have to support verticals directly. Having an interface for the extension library the devs could do what they want there. Also since the adaptive card actions are now added maybe we could show some buttons from there as well in the toolbar (maybe context dependent - which resultwebpart or filterwebpart is active).

The Urls to the webparts could be used as Vertical Type or just as an anchor.

**Comment by wobba** [comment]
Would you be able to create it as a custom suggestions provider somehow? https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_suggestions_provider/ Then we could host it as a separate solution to be downloaded.

**Comment by edarroudi** [comment]
It probably would be somehow possible through the onInit() & domElement reference. But it feels like misusing the interface and practically there is not much difference to the custom webpart approach I have right now.

**Reply by wobba** [reply]
How much bundle size would the code add if we included it in the current search box? Another solution is to offer a separate package with "Advanced search box". And how does your UX work with the suggestion extension?

**Reply by FranckyC** [reply]
@edarroudi, in your case having an independent, custom end-to-end Search Box Web Part that connect with Search Results Web Part with dynamic data feature would be a better option in my opinion. What are exactly the features your customizations provide? If they are too specific to your customer, we probably won't integrate them so they have to be generic.

**Reply by edarroudi** [reply]
I'll think about this in a more generic context and come back to this thread later as I'm busy right now.
